src/plugins/kibana/server/routes/api/settings/register_set.js

```diff
@@ -0,0 +1,33 @@
+import Boom from 'boom';
+import { assign } from 'lodash';
+import defaultsProvider from './defaults';
+
+export default function registerSet(server) {
+  server.route({
+    path: '/api/kibana/settings/{key}',
+    method: 'POST',
```

Comment:
If you expand this to `['POST', 'DELETE']` you can pivot your handler logic based on the request method, or you could just create a new route.